### DISCUSSION TITLE
Config PRG to work with base address other than D400

### DISCUSSION THREAD

**Discussion by insanity213** [frntc/SIDKick-pico#38] [Ideas]
I've wired one of these into a second socket that uses D420 as the chip select on pin 8 and can no longer use the built in config tool as a result.  Would it be possible to publish a stand alone PRG that is coded to use D420 as the base address instead of D400?  If we wanted to get fancy, could we scan on startup for available SIDKicks in a system or maybe just ask for the address of the device you want to configure?

**Comment by frntc** [comment]
The config tool probably doesn't start anymore as the firmware assumes the SID to be at $d400. What board is that? Does it only send the CS for $D420 to $D43F? (in any case the required changes are quite substantial as they affect lots of code parts)

**Reply by insanity213** [reply]
Yeah, I kind of figured the tool assumes D400 for the base of everything.  This is a custom board I'm wiring up by hand.  In the C64 itself I've broken out the d400, d500, d600 and d700 pages.  I use those page's CS pins to drive an additional decoder which breaks down to xx20, xx40 and xx60.  I also have provisions to select de00 or df00 for the base address of my boards decoder if using on a machine that doesn't have d400-d700 broken out.  So it would be nice to select the base address of the Pico on the fly when launching the config tool.  If the source or an uncompressed binary were available, I'd hack something together but it looks like theres only the compressed hex available in the repo thats built into the firmware at compile time.
